### PR TITLE
NodeJs: reorg api response type files

### LINKED ISSUE #2078: NodeJs binding: improve dir/file structure for API types
## Description

Currently we have a lot of files that contain only one type. We should reduce those and logically group the types they contain, e.g. by route group.

## Motivation

* reduces the amount of files 
* it's easier to find stuff

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

Brings the file structure in line with the V3 Core API route groups.

## Links to any relevant issues

Fixes #2078

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/nodejs/lib/client/client.ts [hunk 1]
```diff
@@ -67,14 +67,14 @@ import {
     OutputWithMetadataResponse,
     NetworkMetricsResponse,
 } from '../types/models/api';
-import { RoutesResponse } from '../types/models/api/routes-response';
+import { RoutesResponse } from '../types/models/api/node/routes-response';
```
Comment (DaughterOfMars) [inline bindings/nodejs/lib/client/client.ts]: still `-response`?

Comment (DaughterOfMars) [reply bindings/nodejs/lib/client/client.ts]: Overall confused about the places with/without this suffix

Comment (Alex6323) [reply bindings/nodejs/lib/client/client.ts]: I think I agree with you. I also don't like all those folders now btw. So I'll suggest something different: one file per route group, which means: `accounts.ts`, `blocks.ts`, and so on ... without any suffixes.

Comment (Alex6323) [reply bindings/nodejs/lib/client/client.ts]: I removed all the `-response` suffixes from the files and made the overall structure flat, so all response types live in their respective route group as specified by TIP-48.

File: bindings/nodejs/lib/types/models/api/response.ts [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
-         * The code for the error response.
-         */
-        code: string;
-        /**
-         * A more descriptive version of the error.
-         */
-        message: string;
-    };
-}
+// TODO: can this be removed? Nothing seems to require it :thinking:
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/models/api/response.ts]: What about all the `const parsed = JSON.parse(response) as Response<BasicOutput>;`?

Comment (Thoralf-M) [reply bindings/nodejs/lib/types/models/api/response.ts]: Ah that's something else

Comment (Alex6323) [reply bindings/nodejs/lib/types/models/api/response.ts]: that's using the generic `Response<T> type`, not this one.

Comment (Thoralf-M) [reply bindings/nodejs/lib/types/models/api/response.ts]: Would delete this file :+1:

Comment (Alex6323) [reply bindings/nodejs/lib/types/models/api/response.ts]: Removed

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**